### PR TITLE
REF: Remove Numba compat pre 0.50.1

### PR DESCRIPTION
**Pull request by mroeschke**
No longer needed since the bump https://github.com/pandas-dev/pandas/issues/43257#issuecomment-907452377

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @mroeschke There a still min versions set in `pandas/tests/window/conftest.py`. can these be removed too?

**Comment by mroeschke** [comment]
Thanks. Yes, just removed those as well.

**Comment by simonjayhawkins** [comment]
Thanks @mroeschke

File: pandas/tests/groupby/aggregate/test_numba.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 from pandas.core.util.numba_ import NUMBA_FUNC_CACHE
 
 
-@td.skip_if_no("numba", "0.46.0")
+@td.skip_if_no("numba")
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_numba.py:18]: prob could remove these individual skips and just skip the entire module if numba not installed?

Comment (mroeschke) [reply pandas/tests/groupby/aggregate/test_numba.py:18]: True. All `*/test_numba.py` tests should require numba

**Pull request merged**